### PR TITLE
Tokens page with e2e

### LINKED ISSUE #45: Implement `/tokens` Page
#### What would you like to be added:

```txt
/src/pages/tokens/index.tsx
```

/area tokens
/triage accepted
/assign @siradji

### PR DESCRIPTION
**Pull request by siradji**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
Rought implementation of tokens page  with E2E
#### Which issue(s) does this PR fixes?:

Fixes #45 

#### Screenshots

image

image

image

image

### CODE REVIEW

File: src/interfaces/index.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export interface TokenData {
```
Comment (fuxingloh) [inline src/interfaces/index.ts:1]: access this interface from the package

```ts
import { tokens } from '@defichain/whale-api-client'

tokens.TokenData
```

File: src/components/shared/Loader.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export const Loader = (): JSX.Element => {
```
Comment (fuxingloh) [inline src/components/shared/Loader.tsx:1]: https://github.com/DeFiCh/scan/blob/main/CONTRIBUTING.md#top-level-functions-should-not-use-arrow-syntax

File: src/components/shared/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export * from './Loader'
```
Comment (fuxingloh) [inline src/components/shared/index.ts:1]: don't need to use and maintain `index.ts` since this is not a library project.

**Review by fuxingloh — CHANGES_REQUESTED** [review]

File: src/pages/tokens.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  )
+}
+
+const TokensPage: NextPage = (): JSX.Element => {
+  const [tokens, setTokens] = useState<TokenData[]>([])
+  const [loading, setLoading] = useState<boolean>(true)
+  const [nextToken, setNextToken] = useState<string | number | undefined>(undefined)
+  const [hasNext, setHasNext] = useState<boolean>(false)
+
+  const rpc = useWhaleApiClient()
```
Comment (fuxingloh) [inline src/pages/tokens.tsx:23]:
```suggestion
  const api = useWhaleApiClient()
```

File: src/pages/tokens.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  const [tokens, setTokens] = useState<tokens.TokenData[]>([])
+  const [loading, setLoading] = useState<boolean>(true)
+  const [nextToken, setNextToken] = useState<string | number | undefined>(undefined)
+  const [hasNext, setHasNext] = useState<boolean>(false)
+
+  const api = useWhaleApiClient()
+
+  useEffect(() => {
+    const fetchTokens = async (): Promise<void> => {
+      // NOTE(siradji) Made 2 the default query size. Will be changing it when design is finalized.
```
Comment (canonbrother) [inline src/pages/tokens.tsx:27]:
```suggestion
      // NOTE(siradji): Made 2 the default query size. Will be changing it when design is finalized.
```

File: src/pages/tokens.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  // TODO(siradji) Figure out a way to  implement previous button.
+  const handlePreviousPagination = async (): Promise<void> => {}
+
+  const handleNextPagination = async (): Promise<void> => {
+    if (!hasNext) return
+
+    setLoading(true)
+    const response = await api.tokens.list(2, (nextToken as string))
+    setLoading(false)
+    setTokens(response as any)
```
Comment (canonbrother) [inline src/pages/tokens.tsx:49]: hmm... i think we have proper type, instead of any, can you ensure it?

**Review by canonbrother — COMMENTED** [review]
minor requirement.. can you screenshot the token page?
eg: https://github.com/DeFiCh/scan/pull/62

update:
ohh.. just aware cypress does the screenshot

**Comment by fuxingloh** [comment]
Drop all changes that isn't part of `/tokens`.

**Comment by fuxingloh** [comment]
Also create another PR from main instead, this branch is too stale.

**Pull request closed without merging**